### PR TITLE
update pods provision logic

### PR DESCRIPTION
**Pull request by undassa**

### CODE REVIEW

File: pkg/controller/runtime/deployment/deployment.go [hunk 1]
```diff
@@ -59,10 +59,30 @@ func Provision(d *types.Deployment) error {
 
 	// Check deployment is marked for destroy
 	if d.Spec.State.Destroy {
-		// Mark pod for destroy
+
+		if len(pl) == 0 {
+			log.Debugf("controller:deployment:controller:provision: remove deployment %s", d.SelfLink())
```
Comment (kirillDanshin) [inline pkg/controller/runtime/deployment/deployment.go:64]: I think prefix can be stored in a constant or variable like it was done in other places

File: pkg/node/runtime/cri/docker/events.go [hunk 2]
```diff
@@ -120,17 +129,17 @@ func (r *Runtime) Subscribe(ctx context.Context, state *state.PodState, p chan s
...
 					break
 				}
-				if c == nil {
-					log.Errorf("Container: container not found")
-					break
-				}
 
 				state.SetContainer(container)
+
+
```
Comment (kirillDanshin) [inline pkg/node/runtime/cri/docker/events.go:135]: are this four lines necessary?

**Review by kirillDanshin — CHANGES_REQUESTED** [review]

**Review by unloop — APPROVED** [review]

File: pkg/controller/runtime/deployment/deployment.go (hunk 1 above)
Comment (undassa) [reply pkg/controller/runtime/deployment/deployment.go:64]: Yeah. Do this a bit later, after core code will be migrated.

File: pkg/node/runtime/cri/docker/events.go (hunk 2 above)
Comment (undassa) [reply pkg/node/runtime/cri/docker/events.go:135]: @kirillDanshin what did you mean? Which lines?

Comment (kirillDanshin) [reply pkg/node/runtime/cri/docker/events.go:135]: @undassa I mean empty lines at 134-137

**Pull request merged**